environment.sh

```diff
@@ -36,52 +35,101 @@ done
+  git fetch --all
+  echo "Git branch '$branch_name' exists in the remote repository, pulling latest changes..."
+  git pull origin $branch_name
+fi
 
 # Check if environment exists already
-env_exists=$(conda info --envs | grep $env_name | head -n 1)
-if [[ $env_exists == '' ]]; then
-  env_exists="no"
+create_env=$(conda info --envs | grep $env_name | head -n 1)
+if [[ $create_env == '' ]]; then
+  # No environment exists with this name
+  create_env="yes"
 else
-  env_exists="yes"
+  create_env="no"
 fi
 
-if [ $env_exists == 'yes' ]; then
-  creation_time="$(head -n1 /home/$username/miniconda3/envs/$env_name/conda-meta/history)"
+if [ $create_env == 'no' ]; then
+  # Check if existing environment needs to be updated
   echo "Existing environment found for $env_name."
-  # Check if existing environment is older than a week and remake it if so
+  one_week_ago=$(date -d "7 days ago" '+%Y-%m-%d %H:%M:%S')
+  creation_time="$(head -n1 /home/$username/miniconda3/envs/$env_name/conda-meta/history)"
+  # Check if existing environment is older than a week and delete it if so
   if [[ $one_week_ago > $creation_time ]]; then
     echo "Environment is older than one week old. Deleting and remaking environment...."
     conda remove -n $env_name --all -y
-    env_exists="no"
+    create_env="yes"
   fi
+fi
 
-# Create new environment if it did not exist or we need to remake it
-if [ $env_exists == 'no' ]; then
-# This cannot be an else about because we set env_exists to no in that block
+if [ $create_env == 'yes' ]; then
   # Create new environment
   echo "Environment $env_name does not exist. Creating new environment $env_name..."
   # Create conda environment
   conda create -n $env_name python=3.11 -y
-  # Determine which requirements.txt to install from
-  if [ $env_type == 'simulation' ]; then
-    echo "Installing requirements for simulation environment..."
-    install_file="requirements.txt"
-  elif [ $env_type == 'artifact' ]; then
-    echo "Installing requirements for artifact environment"
-    install_file="artifact_requirements.txt"
-  else
-    echo "Invalid environment type. Valid argument types are simulation and artifact."
-    return 
-  fi
 fi
 
-# Activate new environment
+# Activate environment
 conda activate $env_name
+if [ $create_env == 'yes' ]; then
+  # Install json parser
+  conda install jq -y
+  update_env='yes'
+else
+  update_env='no'
+fi
+
+# At this point we have either a new environment or an environment that is less than a week old.
+# We need to check if we need to reinstall packages
+if [ $update_env == 'no' ]; then
+  # Check if environment was build before last modification to requirements file
+  if [[ $creation_time < $requirements_modification_time ]]; then
+    echo "Requirements file last modified after environment was created. Reinstalling packages..."
+    update_env="yes"
+  fi
+fi
+
+# Check if there has been an update to vivarium packages since last modification to requirements file
```

Comment:
I think we should do these checks in the same place we're checking if the environment is over 7 days old and delete and re-create the environment in that case as well, rather than just modifying it in place.

Reply:
We can do this, however I will point out that if we do not update the requirements.txt for awhile but there are changes to vivarium packages (e.g. we are on a long standing release candidate branch like now) we will essentially be remaking an environment every time we activate. This is why I was just reinstalling the packages instead of remaking the environment since that goes a lot faster.

This is also why I had different if blocks so we could separate these out.